src/Media.Plugin.iOS/PhotoLibraryAccess.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreImage;
+using Foundation;
+using Photos;
+
+namespace Plugin.Media
+{
+    public class PhotoLibraryAccess
+    {
+        public static async Task<NSDictionary> GetPhotoLibraryMetadata(NSUrl url)
+        {
+            NSDictionary meta = null;
+            var completed = false;
+            var options = new PHContentEditingInputRequestOptions
+            {
+                NetworkAccessAllowed = true,
+            };
+            NSUrl[] urls = { url };
+            var image = PHAsset.FetchAssets(urls, new PHFetchOptions()).firstObject as PHAsset;
+            await Task.Run(() =>
```

Comment:
after review is finished but before (I still hope for) merging I'd rebase so there are no unnecessary commits with rejected ideas.